src/core/service.c

```diff
@@ -2705,99 +2708,175 @@ static void service_enter_reload_by_notify(Service *s) {
 
         service_set_state(s, SERVICE_RELOAD_NOTIFY);
```

Comment:
hmm, the state is called SERVICE_RELOAD_NOTIFY but you now enter it via a function called "propagate"? the previous name of the func matched the state name, which makes a lot of sense to me.